hoomd/md/EvaluatorPairTWF.h

```diff
@@ -0,0 +1,242 @@
+                sigma(sigma),
+                alpha(alpha),
+                prefactor(4 * epsilon / (alpha * alpha)) {}
+
+            pybind11::dict asDict()
+                {
+                pybind11::dict v;
+                v["sigma"] = sigma;
+                v["alpha"] = alpha;
+                v["epsilon"] = alpha * alpha / 4 * prefactor;
+                return v;
+                }
+            #endif
+            }
+            __attribute__((aligned(16)));
+
+        //! Constructs the pair potential evaluator
+        /*! \param _rsq Squared distance between the particles
+            \param _rcutsq Squared distance at which the potential goes to 0
+            \param _params Per type pair parameters of this potential
+        */
+        DEVICE EvaluatorPairTWF(Scalar _rsq,
+                                Scalar _rcutsq,
+                                const param_type& _params)
+            : rsq(_rsq), rcutsq(_rcutsq), params(_params)
+            {
+            }
+
+        //! TWF doesn't use diameter
+        DEVICE static bool needsDiameter() { return false; }
+
+        //! Accept the optional diameter values
+        /*! \param di Diameter of particle i
+            \param dj Diameter of particle j
+        */
+        DEVICE void setDiameter(Scalar di, Scalar dj) { }
+
+        //! TWF doesn't use charge
+        DEVICE static bool needsCharge() { return false; }
+
+        //! Accept the optional diameter values
+        /*! \param qi Charge of particle i
+            \param qj Charge of particle j
+        */
+        DEVICE void setCharge(Scalar qi, Scalar qj) { }
+
+        //! Evaluate the force and energy
+        /*! \param force_divr Output parameter to write the computed force divided by r.
+            \param pair_eng Output parameter to write the computed pair energy
+            \param energy_shift If true, the potential must be shifted so that
+            V(r) is continuous at the cutoff
+
+            \return True if they are evaluated or false if they are not because
+            we are beyond the cutoff
+        */
+        DEVICE bool evalForceAndEnergy(Scalar& force_divr,
+                                       Scalar& pair_eng,
+                                       bool energy_shift)
+            {
+            // compute the force divided by r in force_divr
+            if (rsq < rcutsq)
+                {
+                // Compute common terms to equations
+                Scalar sigma2 = params.sigma * params.sigma;
+                // If particles are overlapping return maximum possible energy
+                // and force since this is an invalid state and should be
+                // infinite energy and force.
+                if (rsq <= sigma2) {
+                    // For now use an arbitary constant since
+                    // std::numeric_limit<>::max cannot be used for GPU code.
+                    pair_eng = 1e37;
```

Comment:
You could use NaN explicitly for invalid configurations: https://stackoverflow.com/questions/15514286/way-to-get-floating-point-special-values-in-cuda

Reply:
I have moved to the `INFINITY` macros.